### PR TITLE
Add toolbar to dashboard tiles

### PR DESCRIPTION
**Pull request by NikkiMeganMoore**

### CODE REVIEW

File: src/components/Dashboard/Dashboard.js [hunk 1]
```diff
@@ -78,6 +80,8 @@ class Dashboard extends React.Component {
     isEditing: false,
     notExecutedText: undefined,
     enableDynamicCharting: true,
+    onErrorCallback: () => {},
+    onSuccessCallback: () => {},
     // onChange: () => {}
```
Comment (savithChata) [inline src/components/Dashboard/Dashboard.js:85]: remove?

File: src/components/OptionsToolbar/OptionsToolbar.js [hunk 2]
```diff
@@ -0,0 +1,639 @@
...
+    onSuccessAlert: () => {},
+    onErrorCallback: () => {},
+    onNewNotificationCallback: () => {},
+    deleteMessageCallback: () => {},
+    onFilterCallback: () => {},
+  }
+
+  state = { isHideColumnsModalVisible: false, isSettingColumnVisibility: false }
+
+  // renderInterpretationTip = () => {
```
Comment (savithChata) [inline src/components/OptionsToolbar/OptionsToolbar.js]: remove?

File: src/components/OptionsToolbar/OptionsToolbar.js [hunk 3]
```diff
@@ -0,0 +1,639 @@
...
+    const queryOutputId = _get(this.props.responseRef, 'COMPONENT_KEY')
+
+    try {
+      const filterHeaderElements = document.querySelectorAll(
+        `#chata-response-content-container-${queryOutputId} .chata-table .tabulator-header-filter`
+      )
+      const colHeaderElements = document.querySelectorAll(
+        `#chata-response-content-container-${queryOutputId} .chata-table .tabulator-col`
+      )
+      // const messageElement = document.querySelector(
```
Comment (savithChata) [inline src/components/OptionsToolbar/OptionsToolbar.js]: remove here and the commented out lines below?

File: src/components/OptionsToolbar/OptionsToolbar.js [hunk 4]
```diff
@@ -0,0 +1,639 @@
...
+              <Icon type="download" /> Download as PNG
+            </li>
+          )}
+          {shouldShowButton.showCopyButton && (
+            <li
+              onClick={() => {
+                this.setState({ activeMenu: undefined })
+                this.copyTableToClipboard()
+              }}
+              // className={`chata-toolbar-btn${
```
Comment (savithChata) [inline src/components/OptionsToolbar/OptionsToolbar.js]: remove?

File: src/components/OptionsToolbar/OptionsToolbar.js [hunk 5]
```diff
@@ -0,0 +1,639 @@
...
+    document.body.appendChild(el)
+    el.select()
+    document.execCommand('copy')
+    document.body.removeChild(el)
+    this.setTemporaryState('copiedSQL', true, 1000)
+    this.props.onSuccessAlert(
+      'Successfully copied generated query to clipboard!'
+    )
+    ReactTooltip.hide()
+    // changeTooltipText(
```
Comment (savithChata) [inline src/components/OptionsToolbar/OptionsToolbar.js]: remove?

File: src/components/OptionsToolbar/OptionsToolbar.js [hunk 6]
```diff
@@ -0,0 +1,639 @@
...
+  }
+
+  // todo: put all right toolbar functions into separate component
+  copyTableToClipboard = () => {
+    if (this.props.responseRef) {
+      this.props.responseRef.copyTableToClipboard()
+      this.props.onSuccessAlert('Successfully copied table to clipboard!')
+      this.setTemporaryState('copiedTable', true, 1000)
+      ReactTooltip.hide()
+      // changeTooltipText(
```
Comment (savithChata) [inline src/components/OptionsToolbar/OptionsToolbar.js]: remove?

File: src/components/OptionsToolbar/OptionsToolbar.js [hunk 7]
```diff
@@ -0,0 +1,639 @@
...
+              <Icon type="copy" /> Copy table to clipboard
+            </li>
+          )}
+          {shouldShowButton.showSQLButton && (
+            <li
+              onClick={() => {
+                this.setState({ activeMenu: undefined })
+                this.copySQL()
+              }}
+              // className={`chata-toolbar-btn${
```
Comment (savithChata) [inline src/components/OptionsToolbar/OptionsToolbar.js]: remove?

File: src/components/OptionsToolbar/OptionsToolbar.js [hunk 8]
```diff
@@ -0,0 +1,639 @@
...
+      showCopyButton:
+        isTableResponse(response, displayType) &&
+        !!_get(response, 'data.data.rows.length'),
+      showSaveAsCSVButton:
+        isTableResponse(response, displayType) &&
+        !!_get(response, 'data.data.rows.length'),
+      showSaveAsPNGButton: CHART_TYPES.includes(displayType),
+      showHideColumnsButton:
+        this.props.autoQLConfig.enableColumnVisibilityManager &&
+        // !isAggregation(response) &&
```
Comment (savithChata) [inline src/components/OptionsToolbar/OptionsToolbar.js:471]: remove?

Comment (NikkiMeganMoore) [reply src/components/OptionsToolbar/OptionsToolbar.js:471]: this is for the feature I mentioned in slack. I'll be enabling it soon so want to keep

Comment (savithChata) [reply src/components/OptionsToolbar/OptionsToolbar.js:471]: ohh okay. sounds good

**Pull request merged**